### PR TITLE
Migrate to 12.0 :camel:

### PR DESCRIPTION
**Pull request by ivy-rew**
A friendly conversion provided by market-up2date-keeper :robot: :handshake:

### CODE REVIEW

File: master-detail-demo/.settings/ch.ivyteam.ivy.designer.prefs [hunk 1]
```diff
@@ -1,5 +1,5 @@
-ch.ivyteam.ivy.designer.preferences.DataClassPreferencePage\:DEFAULT_DATA_CLASS=com.axonivy.demo.Data
-ch.ivyteam.ivy.designer.preferences.DataClassPreferencePage\:DEFAULT_NAMESPACE=com.axonivy.demo
-ch.ivyteam.ivy.project.preferences\:PRIMEFACES_VERSION=11
-ch.ivyteam.ivy.project.preferences\:PROJECT_VERSION=100000
-eclipse.preferences.version=1
+ch.ivyteam.ivy.designer.preferences.DataClassPreferencePage\:DEFAULT_NAMESPACE=com.axonivy.demo
+ch.ivyteam.ivy.project.preferences\:PORTAL_VERSION=10
```
[the remark this answers is not in the document]
Comment (phhung-axonivy) [reply master-detail-demo/.settings/ch.ivyteam.ivy.designer.prefs:2]: Hi @ivy-rew 
I imported a project with portal-component dependency into the designer 12.0.0 and converted this project to 12. We see that this project has been converted to use Portal. In the log file, the Portal version 10 is noted. In addition, there is no frame-12 option when creating a new HTML dialog on the designer 12.
Could we ignore them and continue to use frame-10 for the ivy 12?

[the remark this answers is not in the document]
Comment (ivy-rew) [reply master-detail-demo/.settings/ch.ivyteam.ivy.designer.prefs:2]: Hi @phhung-axonivy 
Yes, It should be fine to keep frame-10, as there is no 12 variant.
I have seen this PORTAL_VERSION preference for the first time. Maybe we can just delete it?
To me the dependency to Portal should only be stated in pom.xml files, and nowhere else.

File: master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductDetail/ProductDetailProcess.p.json [hunk 2]
```diff
@@ -64,7 +68,9 @@
       "visual" : {
         "at" : { "x" : 224, "y" : 64 }
       },
-      "connect" : { "id" : "f2", "to" : "f1" }
+      "connect" : [
+        { "id" : "f2", "to" : "f1" }
+      ]
     }, {
       "id" : "f8",
```
[the remark this answers is not in the document]
Comment (ivy-rew) [reply master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductDetail/ProductDetailProcess.p.json:75]: I'd verify whether this problem already occured in the 10.x version.
At any rate, please involve the original developers for questions about the desired functionality.
Unfortunately the master-detail view is a blind spot for me where I was not involved so far.

[the remark this answers is not in the document]
Comment (phhung-axonivy) [reply master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductDetail/ProductDetailProcess.p.json:75]: @ntqdinh-axonivy
This process should be tested by clicking the edit product.

File: master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductList/ProductListData.d.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
+{
```
[the remark this answers is not in the document]
Comment (phhung-axonivy) [reply master-detail-demo/src_hd/com/axonivy/demo/masterdetail/ProductList/ProductListData.d.json:1]: They are info so that we can ignore them.

**Review by ntqdinh-axonivy — APPROVED** [review]
LGTM

**Pull request merged**